### PR TITLE
modify language model example after change ptb dataset to sherlockholmes

### PR DESCRIPTION
**Pull request by liyujiel**
## Description ##
There is an example in the Gluonnlp repo that uses a copy of the Penn Treebank (PTB) hosted at https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/ptb. We should not be redistributing that corpus so wants to remove it from gihubusercontent. 

I update self-created dataset named Sherlock Holmes in web-data
Also, the related notebook is tested and modified.

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [x] Data Load Method 
- [x] Language model example

### CODE REVIEW

**Comment by mli** [comment]
Job PR-174/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-174/2/index.html

File: docs/examples/language_model/get_sherlockholmes_data.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+CURR_DIR=$(cd `dirname $0`; pwd)
+
+wget -P ${CURR_DIR} https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/sherlockholmes/sherlockholmes.train.txt;
```
Comment (szha) [inline docs/examples/language_model/get_sherlockholmes_data.sh]: I don't have wget on mac. Shall we use the download function in mxnet instead?

Comment (liyujiel) [reply docs/examples/language_model/get_sherlockholmes_data.sh]: sounds good, changed to download and remove shell script. 👍

**Comment by mli** [comment]
Job PR-174/3 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-174/3/index.html

File: docs/examples/language_model/language_model.ipynb [hunk 2]
```diff
@@ -407,20 +408,29 @@
...
    "source": [
-    "!./get_ptb_data.sh\n",
-    "ptb_dataset = !ls ptb.*.txt\n",
-    "print(ptb_dataset)"
+    "TRAIN_PATH = \"./sherlockholmes.train.txt\"\n",
+    "VALID_PATH = \"./sherlockholmes.valid.txt\"\n",
+    "TEST_PATH = \"./sherlockholmes.test.txt\"\n",
+    "PREDICT_PATH = \"./tinyshakespeare/input.txt\"\n",
+    "download(\n",
+    "        \"https://raw.githubusercontent.com/dmlc/web-data/master/mxnet/sherlockholmes/sherlockholmes.train.txt\",\n",
```
Comment (szha) [inline docs/examples/language_model/language_model.ipynb]: do you have the hash handy? we can use it to check the file integrity (and handle failed/partial downloads)

**Comment by mli** [comment]
Job PR-174/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-174/4/index.html

File: docs/examples/language_model/language_model.ipynb (hunk 2 above)
Comment (liyujiel) [reply docs/examples/language_model/language_model.ipynb]: Thanks for pointing me out, maybe we should add checksum with all external(customized) data resources. :)

**Pull request merged**